packages/core-database/src/database-service-factory.ts

```diff
@@ -1,12 +1,22 @@
 import { Database } from "@arkecosystem/core-interfaces";
 import { DatabaseService } from "./database-service";
-import { DelegatesRepository } from "./repositories/delegates";
-import { WalletsRepository } from "./repositories/wallets";
+import { BlocksBusinessRepository } from "./repositories/blocks-business-repository";
+import { DelegatesBusinessRepository } from "./repositories/delegates-business-repository";
+import { TransactionsBusinessRepository } from "./repositories/transactions-business-repository";
+import { WalletsBusinessRepository } from "./repositories/wallets-business-repository";
 
 // Allow extenders of core-database to provide, optionally, a IWalletManager concrete in addition to a IDatabaseConnection, but keep the business repos common
 export const databaseServiceFactory = async (opts: any, walletManager: Database.IWalletManager, connection: Database.IDatabaseConnection): Promise<Database.IDatabaseService> => {
     let databaseService: DatabaseService;
-    databaseService = new DatabaseService(opts, connection, walletManager, new WalletsRepository(() => databaseService), new DelegatesRepository(() => databaseService));
+    databaseService = new DatabaseService(
```

Comment:
This should be simplified rather then passing in more and more arguments for every repository that gets added.

_Any specific reason other then easier mocking that you pass it to the constructor as an argument rather then initialising it directly in the constructor itself?_

Reply:
This is just a DI(dependency injection) convention. The factory method is just an opinionated way of constructing the database-service instance, not a decree.
We want to retain the ability to pass in different business repo implementations. 
This is in-line with "code is open to extension, closed to modification" methodology.
If the dependencies get too many(don't think it has yet), it's a sign the class is doing too much and needs to be broken up